src/js/data-model.jsx

```diff
@@ -1,82 +1,153 @@
-
-        var items = _.map(entities, function(item) {
-            // Check if the item is the selected resource.
-            var className = item.ident === resource ? 'active' : '',
-                url = router.reverse(item.domain, item.ident);
+    getDefaultProps: function() {
+        return {
+            description: '',
+            url: "#",
+        };
+    },
 
+    getInitialState: function() {
+        return {
+            collapsedStatus: {},
+            tableWidth:600,
+        };
+    },
+    
+    render: function() {
+        var self = this;
+
+        var description = this.props.description;
+        if (this.props.tables && this.props.tables.length>0 && self.props.activeTable) {
+            var table = _.find(this.props.tables, function(f) { return f.name === self.props.activeTable; });
+            if(self.props.activeField) {
+                description = _.find(table.fields, function(f) { return f.name === self.props.activeField; }).description;
+                self.state.collapsedStatus[table.name] = false; // if a field is selected, the table shouldn't be collapsed
+            }
+            else {
+                description = table.description;
+            }
+        }
+        
+        var tables = _.map(this.props.tables, function(table) {
+            var isActiveTable = (table.name === self.props.activeTable);
             return (
-                <li key={item.ident} className={className}>
-                    <a href={url}>{item.label}</a>
+                <li key={table.name}>
+                    <Table model={self.props.name} 
+                       version={self.props.version}
+                       name={table.name}
+                       fields={table.fields}
+                       isActive = {isActiveTable} 
+                       activeField = {isActiveTable? self.props.activeField : ''}
+                       isCollapsed = {self.isTableCollapsed(table.name)}
+                       handleClick = {self.toggleCollapsed(table.name)}/>
                 </li>
             );
         });
-
-        var details, entity, object, log;
-
-        if (this.props.resource) {
-            entity = entities.index[this.props.resource];
-
-            if (entity) {
-                object = <DataModelObject {...entity} />;
-
-                details = (
-                    <div>
-                        {object}
-
-                        <hr />
-
-                        <h4>Log</h4>
-
-                        <FactLog facts={entity.facts} comparator={Comparators.time} />
-                    </div>
-                );
-            }
-        }
-
+        
+        var dataDictElement = this.props.siteComments ?
+            <li key='data-dict'>
+                <CollapsibleText title={'Site Comments [Implementation Status: ' + this.props.siteComments.status + ']'} 
+                                  content={this.props.siteComments.comment}/>
+            </li>
+            : null;
```

Comment:
I am not a huge fan of the ternary operator except for very short inline uses. I think these set of declarations would be more clear by used initializing the variables and then just uses `if` blocks.

```javascript

var dataDictElement,
    descriptionElement,
    etlConventionsElement,
    dqaElement;

if (this.props.siteComments) {
    dataDictElement = ...
}
```

I know it is a tiny bit more verbose, but it took me a minute to understand why the ternary expression was being used.